### PR TITLE
Correctly merge pod target xcconfigs for pod target build settings

### PR DESCRIPTION
**Pull request by dnkoutso**
closes https://github.com/CocoaPods/CocoaPods/issues/7798

### CODE REVIEW

File: spec/unit/target/build_settings/pod_target_settings_spec.rb [hunk 1]
```diff
@@ -233,17 +233,12 @@ class BuildSettings
...
             @banana_pod_target = fixture_pod_target(@banana_spec)
 
+            @matryoshka_spec = fixture_spec('matryoshka/matryoshka.podspec')
+            @matryoshka_pod_target = fixture_pod_target_with_specs([@matryoshka_spec, *@matryoshka_spec.subspecs])
+
             @coconut_spec = fixture_spec('coconut-lib/CoconutLib.podspec')
             @coconut_test_spec = @coconut_spec.test_specs.first
             @coconut_pod_target = fixture_pod_target_with_specs([@coconut_spec, @coconut_test_spec])
-
-            @consumer = @coconut_pod_target.spec_consumers.first
```
Comment (dnkoutso) [inline spec/unit/target/build_settings/pod_target_settings_spec.rb]: none of these were needed or used.

**Comment by CocoaPodsBarista** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please include a CHANGELOG entry to credit yourself! <br />
You can find it at <a href="https://github.com/CocoaPods/CocoaPods/blob/master/CHANGELOG.md">CHANGELOG.md</a>.</td>
    </tr>
  </tbody>
</table>

Here's an example of your CHANGELOG entry:

```markdown
* Correctly merge pod target xcconfigs for pod target build settings  
  [dnkoutso](https://github.com/dnkoutso)
  [#issue_number](https://github.com/CocoaPods/CocoaPods/issues/issue_number)
```

*note*: There are two invisible spaces after the entry's text.


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: lib/cocoapods/target/build_settings.rb [hunk 2]
```diff
@@ -489,7 +486,7 @@ def initialize(target, test_xcconfig)
         # @return [Xcodeproj::Xconfig]
         define_build_settings_method :xcconfig, :memoized => true do
           xcconfig = super()
-          xcconfig.merge(pod_target_xcconfig)
+          xcconfig.merge(merged_pod_target_xcconfigs)
```
Comment (dnkoutso) [inline lib/cocoapods/target/build_settings.rb]: `merged_pod_target_xcconfigs` was unused. With this change `pod_target_xcconfig` is now unused so it's deleted. Not sure if it was copy pasta or what the intention was here and `pod_target_xcconfig_values_by_consumer_by_key` was never invoked.

File: spec/unit/target/build_settings/pod_target_settings_spec.rb [hunk 3]
```diff
@@ -254,6 +249,14 @@ class BuildSettings
             xcconfig.to_hash['GCC_PREPROCESSOR_DEFINITIONS'].should == '$(inherited) COCOAPODS=1 NON_TEST_FLAG=1'
           end
 
+          it 'merges pod target xcconfig settings from subspecs' do
```
Comment (dnkoutso) [inline spec/unit/target/build_settings/pod_target_settings_spec.rb]: verified the same test passes on 1.5.3 but not in 1.6.0 (as of 41b56a5d91c2d9e66a29d2dec06b20d8aba71447)

File: lib/cocoapods/target/build_settings.rb [hunk 4]
```diff
@@ -786,25 +783,14 @@ def requires_objc_linker_flag?
           )
         end
 
-        # @return [Hash<String => String>]
-        define_build_settings_method :pod_target_xcconfig, :memoized => true do
```
Comment (dnkoutso) [inline lib/cocoapods/target/build_settings.rb]: no longer used after this change

**Comment by amorde** [comment]
Nice!

**Pull request merged**